core/src/main/scala/neotypes/DeferredQuery.scala

```diff
@@ -8,63 +8,187 @@ import scala.collection.compat.Factory
+    * @param rm result mapper for type T
+    * @tparam F effect type
+    * @return F[T]
+    */
   def single[F[_]](session: Session[F])
                   (implicit F: Async[F], rm: ResultMapper[T]): F[T] =
     session.transact(tx => single(tx))
 
+  /** Evaluate query for effect type F in session transaction
+    *
+    * @param session neotypes session for effect type F
+    * @param F effect type F
+    * @param rm result mapper for type T
+    * @tparam F effect type
+    * @return F[T]
+    */
   def execute[F[_]](session: Session[F])
                    (implicit F: Async[F], rm: ExecutionMapper[T]): F[T] =
     session.transact(tx => execute(tx))
 
+  /** List as results collection
+    *
+    * @param tx neotypes transaction for effect type F
+    * @param F effect type F
+    * @param rm result mapper for type T
+    * @tparam F effect type
+    * @return F[List[T]]
+    */
   def list[F[_]](tx: Transaction[F])
                 (implicit F: Async[F], rm: ResultMapper[T]): F[List[T]] =
     tx.list(query, params)
 
+  /** Map as results collection
+    *
+    * @param tx neotypes transaction for effect type F
+    * @param ev  Map (Key, Value) type constraint.  T must be subtype of (K, V)
+    * @param F effect type F
+    * @param rm result mapper for type (K, V)
+    * @tparam F effect type
+    * @tparam K Key for Map
+    * @tparam V Value for Map
+    * @return F[Map[K, V]]
+    */
   def map[F[_], K, V](tx: Transaction[F])
                      (implicit ev: T <:< (K, V), F: Async[F], rm: ResultMapper[(K, V)]): F[Map[K, V]] =
     tx.map(query, params)
 
+  /** Set as results collection
+    *
+    * @param tx neotypes transaction for effect type F
+    * @param F effect type F
+    * @param rm result mapper for type T
+    * @tparam F effect type
+    * @return F[Set[T]]
+    */
   def set[F[_]](tx: Transaction[F])
                (implicit F: Async[F], rm: ResultMapper[T]): F[Set[T]] =
     tx.set(query, params)
 
+  /** Vector as results collection
+    *
+    * @param tx neotypes transaction for effect type F
+    * @param F effect type F
+    * @param rm result mapper for type T
+    * @tparam F effect type
+    * @return F[Vector[T]]
+    */
   def vector[F[_]](tx: Transaction[F])
                   (implicit F: Async[F], rm: ResultMapper[T]): F[Vector[T]] =
     tx.vector(query, params)
 
+  /** First record in the result
+    *
+    * @param tx neotypes transaction for effect type F
+    * @param F effect type F
+    * @param rm result mapper for type T
+    * @tparam F effect type
+    * @return F[T]
+    */
   def single[F[_]](tx: Transaction[F])
                   (implicit F: Async[F], rm: ResultMapper[T]): F[T] =
     tx.single(query, params)
 
+  /** Evaluate query for effect type F
+    *
+    * @param tx neotypes transaction for effect type F
+    * @param F effect type F
+    * @param rm result mapper for type T
+    * @tparam F effect type
+    * @return F[T]
+    */
   def execute[F[_]](tx: Transaction[F])
```

Comment:
@BalmungSan Does the actual execution of a statement depend on when (for example in Cats Effect) an unsafe operation triggers the side effect to occur?  In that case is it impossible to know when the actual effect will be executed?  I'm having difficulty describing what an operation like execute performs when (I think) the output of this method is a description of the side effect to be evaluated, not the execution itself.

Reply:
I wouldn't bother too much with when the real execution is happening, but rather on what returned **effect** describes. In this case, as I said in the other comment, it is:

> Executes the query and ignores its output.

Reply:
@BalmungSan Sounds good, made the changes with examples